### ISSUE TITLE
Images not recomposing after changing the list of a LazyList

### ISSUE THREAD

**Issue by davidsal** [skydoves/Landscapist#28]
**Please complete the following information:**
- Glide and Fresco v1.2.4
 
**Describe the Bug:**

I'm observing a Live Data List using observeAsState(). When I trigger a new event using value/postDelayed, the LazyList (LazyRow/LazyColumn) item recomposes everything (text, icons, colors) except the Glide/Fresco image, making the item keep the old image URL. I'm attaching a small video, you can see that when I remove an item for the list, the title and rating updates, but the image doesn't. This doesn't happen if I use the classic  Accompanist Glide library. 
[WhatsApp Video 2021-06-21 at 22.41.zip](https://github.com/skydoves/Landscapist/files/6691056/WhatsApp.Video.2021-06-21.at.22.41.zip)

**Expected Behavior:**

Whenever the list state changes, the image should recompose too.

**Current Glide Code:**

```
>@Composable
> fun PosterImage(posterURL:String){
>     GlideImage(
>                 imageModel = posterURL,
>                 contentScale = ContentScale.FillWidth,
>                 circularRevealedEnabled = true,
>                 placeHolder = ImageBitmap.imageResource(R.drawable.entry_placeholder),
>                 alignment = Alignment.Center,
>                 modifier = Modifier
>                     .fillMaxWidth()
>                     .height(170.dp),
>                 requestOptions = RequestOptions().diskCacheStrategy(DiskCacheStrategy.AUTOMATIC)
>             )
> }
```

**Comment by skydoves** [comment]
Hi @davidsal,
Does the `CoilImage` face the same issue?
I think that seems to happen on the GlideImage only from now, I will check into the requests.
Thank you for your issue :)

**Comment by madhurgupta10** [comment]
@davidsal Same Issue for me, I am using Glide for Firebase Storage Reference!

**Comment by skydoves** [comment]
Hi @madhurgupta10,
Thanks for sharing your issue :)
I'm working on this issue and it will be included in the next release probably today.
Thanks!

**Comment by skydoves** [comment]
@davidsal @madhurgupta10 
Released a new stable `1.2.8`!

**Comment by madhurgupta10** [comment]
@skydoves Thanks a lot, the issue is fixed for me :)
